float64 accuracy
float64 repeatability
